### PR TITLE
Define REST API to GET results of a promotion assoc with the deployment

### PR DESCRIPTION
**Pull request by maheshkelkar**
References #95

### CODE REVIEW

File: features/api/deployment/updating.feature [hunk 1]
```diff
@@ -4,6 +4,7 @@ Feature: Deployment UPDATE APIs
   I should be able to update existing deployments in the system
 
 
+  @freezetime @wip
   Scenario: Updating a deployment with a status STARTED
```
Comment (GiriDandu) [inline features/api/deployment/updating.feature]: Can you please be more expressive. What is the reason for this "update"

Comment (GiriDandu) [inline features/api/deployment/updating.feature]: Can you please be more expressive. State the reason for updating, why did we get this trigger

File: features/api/deployment/updating.feature [hunk 2]
```diff
@@ -13,9 +14,35 @@ Feature: Deployment UPDATE APIs
         "status" : "COMPLETED"
```
Comment (GiriDandu) [inline features/api/deployment/updating.feature]: Are we updating status with STARTED or COMPLETED?

File: features/api/promotion/reading.feature [hunk 3]
```diff
@@ -46,3 +46,36 @@ Feature: Promotion READ APIs
...
+    Given there is a deployment
+    When I GET "/api/deployments/1/promotions/1"
+    Then the response should be 200
+    And the body should be JSON:
+    """
+      {
+        "id" : 1,
+        "promotion" : "jenkins-smoke",
+        "status" : "STARTED",
+        "infoUrl" : null,
```
Comment (GiriDandu) [inline features/api/promotion/reading.feature]: Why the infoUrl "null" here.  For a success case, should we provide the correct infoUrl?

File: features/api/promotion/reading.feature [hunk 4]
```diff
@@ -46,3 +46,36 @@ Feature: Promotion READ APIs
...
+        "createdAt" : "{{created_timestamp}}"
+      }
+    """
+
+
+  @wip
+  Scenario: Fetching Promotion result for a Deployment that doesn't exist
+
+    When I GET "/api/deployments/1/promotions/1"
```
Comment (GiriDandu) [inline features/api/promotion/reading.feature]: This is deployment that doesn't exist. More interesting case would be deployment is valid but promotion doesn't exist

File: features/api/deployment/updating.feature (hunk 1 above)
Comment (maheshkelkar) [inline features/api/deployment/updating.feature]: This part is already reviewed and approved and not changed in this commit. 

IMO scenario(s) are very specific and succinct. What if I update the feature "description" (at the top) as follows:
The deployDB maintains the status or information about the deployments in central repository. These REST APIs allow the clients to update the status or information of existing deployments and provide up-to-date view for the benefit of administrator.

File: features/api/deployment/updating.feature (hunk 2 above)
Comment (maheshkelkar) [inline features/api/deployment/updating.feature]: fixed the text in scenario

File: features/api/promotion/reading.feature (hunk 3 above)
Comment (maheshkelkar) [inline features/api/promotion/reading.feature]: Currently promotion result is in STARTED state, so I dont have infoUrl. In general infoUrl is optional and can be null

File: features/api/promotion/reading.feature (hunk 4 above)
Comment (maheshkelkar) [inline features/api/promotion/reading.feature]: The next scenario already covers this:

Scenario: Fetching a result for a Promotion which is not associated with the Deployment

File: features/api/deployment/updating.feature (hunk 1 above)
Comment (GiriDandu) [inline features/api/deployment/updating.feature]: Just to make sure I understand this better.  In what circumstance the administrator would change the state of the deployment.  To retry the failed deployment again. It's not clear (at least to me) why you need this during "normal/working"  system

File: features/api/promotion/reading.feature (hunk 3 above)
Comment (GiriDandu) [inline features/api/promotion/reading.feature]: ok, i get it

File: features/api/deployment/updating.feature (hunk 1 above)
Comment (maheshkelkar) [inline features/api/deployment/updating.feature]: Administrator is simply looking at the console and tracking all the deployments. 

The client i.e. orchestrator (rundeck) is updating the status of deployment to report the progress of deployment.

Comment (GiriDandu) [inline features/api/deployment/updating.feature]: OK, I understand now

**Comment by GiriDandu** [comment]
:+1:

**Pull request merged**